curator-client/src/main/java/org/apache/curator/ensemble/exhibitor/ExhibitorEnsembleProvider.java

```diff
@@ -301,7 +301,7 @@ else if ( port != thePort )
                     values.putAll(decodeExhibitorList(encoded));
                     done = true;
                 }
-                catch ( Throwable e )
+                catch ( Exception e )
```

Comment:
NPE still gets caught. If OOM is thrown, does it really make sense to retry? That's usually a pretty drastic condition and very often the JVM is not actually in a recoverable state. The OOM can be thrown because of a different thread going wonky, and we have no way to clean up the resources it was using.